packages/components/src/text/mixins.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * External dependencies
+ */
+import css from '@emotion/css';
+/**
+ * Internal dependencies
+ */
+import { fontFamily } from './font-family';
+
+const fontWeightNormal = `font-weight: 400;`;
+const fontWeightSemibold = `font-weight: 600;`;
+
+const title = `
+  ${ fontWeightNormal }
+`;
+
+const titleLarge = `
+	font-size: 32px;
+	line-height: 40px;
+`;
+
+const titleMedium = `
+	font-size: 24px;
+	line-height: 32px;
+`;
+
+const titleSmall = `
+	font-size: 20px;
+	line-height: 28px;
+`;
+
+const subtitle = `
+	${ fontWeightSemibold }
+	font-size: 14px;
+	line-height: 20px;
+`;
+
+const subtitleLarge = `
+	font-size: 16px;
+	line-height: 24px;
+`;
+
+const subtitleSmall = `
+	font-size: 14px;
+	line-height: 20px;
+`;
+
+const body = `
+	${ fontWeightNormal }
+`;
+
+const bodyLarge = `
+	font-size: 16px;
+	line-height: 24px;
+`;
+
+const bodySmall = `
+	font-size: 14px;
+	line-height: 20px;
+`;
+
+const button = `
+  ${ fontWeightSemibold }
+  font-size: 14px;
+  line-height: 20px;
+`;
+
+const caption = `
+	${ fontWeightNormal }
+	font-size: 12px;
+	line-height: 16px;
+`;
+
+const label = `
+	${ fontWeightSemibold }
+	font-size: 12px;
+	line-height: 16px;
+`;
+
+/**
+ * @typedef {"title.large" | "title.medium" | "title.small" | "subtitle" | "subtitle.small" | "body" | "body.small" | "button" | "caption" | "label"} TextVariant
```

Comment:
It's great that we have this type defined. @aduth clarified coding guidelines around custom types. Can you align with the latest recommendations here? See: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/coding-guidelines.md#custom-types
```suggestion
 * @typedef {'title.large'|'title.medium'|'title.small'|'subtitle'|'subtitle.small'|'body'|'body.small'|'button'|'caption'|'label'} TextVariant
```

Reply:
Spacing isn't important here?

Reply:
The idea was to skip spaces to make it shorter to fit in the line width capped to 80 :)